### ISSUE TITLE
Add S1121 for TS/JS ('no-nested-assignment') 

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#1408]
We allow assignment in loop condition, in eslint it's possible to wrap in parentheses to ignore assignment in condition. I think we should go for this option enabled and w/o ability to configure it in SQ.

**Comment by vilchik-elena** [comment]
`no-cond-assign` reports on conditions only and reports badly (w/o end location) so we would report on entire line which is hard to perceive (btw VSCode in this case highlights only 1 token).

I suggest that we write our own rule with proper reporting and parameter which allow to enable-disable check for assignments inside expression but not in condition:
```
return x = 42; // we are reporting it now in JS, but not in TS
```

Note that in JS we have a lot of exclusions to not report a lot of noise.
